internal/clients/oidc.go

```diff
@@ -0,0 +1,79 @@
+package clients
+
+import (
+	"context"
+	"time"
+
+	"github.com/coreos/go-oidc/v3/oidc"
+	"github.com/ydataai/go-core/pkg/common/logging"
+	"golang.org/x/oauth2"
+)
+
+// OIDCClient defines a struct that can be used
+type OIDCClient struct {
+	configuration OIDCConfiguration
+	OAuth2Config  *oauth2.Config
+	Verifier      *oidc.IDTokenVerifier
+	provider      *oidc.Provider
+	readyzFunc    func() bool
+	logger        logging.Logger
+}
+
+// NewOIDCClient defines a new values for the server
+func NewOIDCClient(logger logging.Logger, config OIDCConfiguration) *OIDCClient {
+	return &OIDCClient{
+		configuration: config,
+		readyzFunc:    func() bool { return false },
+		logger:        logger,
+	}
+}
+
+// Setup initializes setup for OIDC Provider
+func (oc *OIDCClient) Setup() {
+	ctx := context.Background()
+
+	// make sure it is available
+	oc.isAvailable(ctx)
```

Comment:
I didn't get this point. What's happen here if it's not available?
Why do we need to check every 10 seconds?

Reply:
This's only done once until we have a connection to the OIDC provider. Then readyzFunc returns true and that's it.

Reply:
if it's not available, we'll load an http server anyway, but we'll handle the requests to say the server isn't available yet.

Reply:
In that case, doesn't make any sense this verification.